### DISCUSSION TITLE
Is there any other ways to choose outputs in OpenFAST-Simulink?

### DISCUSSION THREAD

**Discussion by Melancho7ee** [OpenFAST/openfast#1867] [General]
I know that it is recommended to use the "Fcn" block in Simulink to choose outputs, such as `u(strmatch('GenPwr',OutList))`. I am just wondering if there is any other ways to choose outputs? Can one use the "MATLAB fcn" block to choose outputs?

**Comment by jjonkman** [comment]
Dear @Melancho7ee,

I'm not sure I understand your question.

The OpenFAST summary file lists the index of each output stored in the `OutList` array, so, you can also access the values directly through those indices.

Best regards,

**Comment by Melancho7ee** [comment]
Thanks for your reply. I am trying to output nodal outputs in AeroDyn, such as outputting from `u(strmatch('AB1N001Fx',OutList))` to `u(strmatch('AB1N019Fx',OutList))`. The total 19 parameters are laborious if I use the `strmatch` methods.
Yesterday I have found a way to solve this problem, shown in the figure below.
Matlab code in the matlab function block is
```
function y = fcn(u,n1)

y = u(n1:n1+18)
```
It is simpler to output the 19 outputs, which is what I wanted when I asked this question.

![CleanShot 2023-11-12 at 09 39 29@2x](https://github.com/OpenFAST/openfast/assets/116873649/b40d223f-a268-4c68-b541-d3c71c767ee3)
